### PR TITLE
Feature/declare over button

### LINKED ISSUE #8: Remove (Declare End) Incident
As a user, I can declare an incident's ending and move it to resolved.

### PR DESCRIPTION
**Pull request by JeffShepherd**
### What Changed?
- Add button to that declares an end to the incident and updates the backend
### What's Next?
- Style application, add tests, and update chart view once the relevant endpoint is complete
### Known Issues?
- N/A
### Test Coverage?
- Test coverage to be added in future efforts

### CODE REVIEW

File: src/components/IncidentInfo/IncidentInfo.js [hunk 1]
```diff
@@ -12,6 +12,17 @@ const IncidentInfo = ({ id }) => {
...
   }, [dispatch])
 
+  const handleSubmit = e => {
+    e.preventDefault()
+    const incidentObj = {
+      active: false,
+      close_date: new Date(Date.now())
+    }
+    console.log(incidentObj)
+    dispatch(updateIncidentInfo({id, incidentObj}))
```
Comment (percworld) [inline src/components/IncidentInfo/IncidentInfo.js:22]: Thanks Cameron for researching this object

File: src/components/IncidentInfo/IncidentInfo.js [hunk 2]
```diff
@@ -22,11 +33,15 @@ const IncidentInfo = ({ id }) => {
         {console.log(incident)}
         <p data-cy='info-name' className='info info-name'>{incident.name}</p>
         <p data-cy='info-type' className='info type'><span className='info-type'>{getIconByType(incident.incident_type)}</span>{incident.incident_type}</p>
-        <a href={mapURL} target="_blank" rel="noopener noreferrer" className='map-btn'>Click to get Map to Headquarters</a>
```
Comment (percworld) [inline src/components/IncidentInfo/IncidentInfo.js:25]: Is this deleted from the code?

File: src/components/IncidentInfo/IncidentInfoSlice.js [hunk 3]
```diff
@@ -16,6 +17,16 @@ export const getIncidentInfo = createAsyncThunk(
   }
 )
 
+export const updateIncidentInfo = createAsyncThunk(
+  'incidentInfo/updateIncidentInfo',
+  async ({id, incidentObj}) => {
+    console.log(id)
```
Comment (cbmackintosh) [inline src/components/IncidentInfo/IncidentInfoSlice.js:23]: If we have time we should review the Redux Toolkit docs to find out exactly what these different arguments in async represent - good to remember for the future though, you can only pass one argument into async. If you want more than one, you need to destructure

File: src/components/IncidentInfo/IncidentInfo.js [hunk 4]
```diff
@@ -12,6 +12,17 @@ const IncidentInfo = ({ id }) => {
...
+  const handleSubmit = e => {
+    e.preventDefault()
+    const incidentObj = {
+      active: false,
+      close_date: new Date(Date.now())
+    }
+    console.log(incidentObj)
+    dispatch(updateIncidentInfo({id, incidentObj}))
+  }
+
```
Comment (cbmackintosh) [inline src/components/IncidentInfo/IncidentInfo.js:24]: Let's not forget to delete all our console logs before project submission

File: src/components/IncidentInfo/IncidentInfo.js (hunk 2 above)
Comment (JeffShepherd) [reply src/components/IncidentInfo/IncidentInfo.js:25]: We're good to go here. I wrapped the new button, and the map button, inside a container so I can flex them into a row.

File: src/components/IncidentInfo/IncidentInfo.js [hunk 5]
```diff
@@ -22,11 +33,15 @@ const IncidentInfo = ({ id }) => {
...
+          <a href={mapURL} target="_blank" rel="noopener noreferrer" className='map-btn'>Click to get Map to Headquarters</a>
+          {!incident.close_date && <button className="end-button" onClick={e => handleSubmit(e)}>Declare Incident Over</button>}
+        </div>
         <p data-cy='info-location' className='info info-location'>Located at: {incident.location} in {incident.city}, {incident.state}</p>
         <p data-cy='info-desc' className='info info-desc'>Summary: {incident.description}</p>
         <p data-cy='info-start-date' className='info info-start-date'>Declaration: {formatDate(incident.start_date)}</p>
-        <p data-cy='info-end-date' className='info info-end-date'>Resolution: {formatDate(incident.close_date)}</p>
+        {console.log(incident.close_date)}
+        {incident.close_date && <p data-cy='info-end-date' className='info info-end-date'>Resolution: {formatDate(incident.close_date)}</p>}
         <div data-cy='info-instructions' className='info info-instructions'>Reminders & Instructions: {getInstructions(incident.incident_type)}</div>
```
Comment (cbmackintosh) [inline src/components/IncidentInfo/IncidentInfo.js:45]: Something else we might do here:
`<p data-cy='info-end-date' className='info info-end-date'>Resolution: {incident.active ? 'Ongoing' : formatDate(incident.close_date)}</p>`

File: src/components/IncidentInfo/IncidentInfo.js [hunk 6]
```diff
@@ -22,11 +33,15 @@ const IncidentInfo = ({ id }) => {
         {console.log(incident)}
         <p data-cy='info-name' className='info info-name'>{incident.name}</p>
         <p data-cy='info-type' className='info type'><span className='info-type'>{getIconByType(incident.incident_type)}</span>{incident.incident_type}</p>
-        <a href={mapURL} target="_blank" rel="noopener noreferrer" className='map-btn'>Click to get Map to Headquarters</a>
+        <div className="info-button-container">
+          <a href={mapURL} target="_blank" rel="noopener noreferrer" className='map-btn'>Click to get Map to Headquarters</a>
+          {!incident.close_date && <button className="end-button" onClick={e => handleSubmit(e)}>Declare Incident Over</button>}
```
Comment (cbmackintosh) [inline src/components/IncidentInfo/IncidentInfo.js:38]: Might be more readable to use incident.active here, like so:
`{incident.active && <button className="end-button" onClick={e => handleSubmit(e)}>Declare Incident Over</button>}`

**Review by cbmackintosh — APPROVED** [review]
Nice work troubleshooting this important piece of functionality. Good to have it squared away.

File: src/components/IncidentInfo/IncidentInfo.js (hunk 5 above)
Comment (percworld) [reply src/components/IncidentInfo/IncidentInfo.js:45]: or Ongoing: formatDate(Date.now)?

**Review by percworld — APPROVED** [review]

**Pull request merged**